### ISSUE TITLE
OpenCv-MacOS: FATAL: Error, problem with dependency scan please report the bug.

### ISSUE THREAD

**Issue by Josh596** [Nuitka/Nuitka#1645]
**Nuitka version**
```
1.0rc3
Commercial: None
Python: 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53) 
Flavor: Unknown
Executable: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/bin/python3
OS: Darwin
Arch: x86_64
```
**How did I install Nuitka and Python**
Nuitka - `pip install -U --force-reinstall git+https://github.com/Nuitka/Nuitka.git@develop` in a virtual environment
Python -  python.org

**python3 -m pip freeze**
```
**Relevant packages**
imageio==2.19.3
importlib-metadata==4.11.4
llvmlite==0.38.0
matplotlib==3.5.2
Nuitka @ git+https://github.com/Nuitka/Nuitka.git@2dcb17e0352beb957f7ea4286b8c306a51d8b0bf
numba==0.55.1
numpy==1.21.6
opencv-contrib-python==4.6.0.66
opencv-python==4.6.0.66
ordered-set==4.1.0
Pillow==9.1.1
pyobjc==8.5
pyobjc-core==8.5
PySide6==6.3.0
PySide6-Addons==6.3.0
PySide6-Essentials==6.3.0
python-dateutil==2.8.2
requests==2.26.0
rubicon-objc==0.4.2
scikit-image==0.19.2
scipy==1.8.1
shiboken6==6.3.0
six==1.16.0
```

**Example**
```
"""A simple script to demonstrate the issue"""

import sys
import cv2
import numpy as np
from PIL import Image
from PySide6.QtCore import Qt
from PySide6.QtWidgets import QApplication, QLabel, QWidget

def main():
    app = QApplication(sys.argv)
    window = QWidget()
    window.setWindowTitle("Simple")
    window.setGeometry(300, 300, 300, 300)
    label = QLabel(window)
    label.setText("Hello World!")
    label.setGeometry(0, 0, 300, 300)
    label.setAlignment(Qt.AlignCenter)
    window.show()

    im = Image.new(mode="RGB", size=(200, 200))
    im_array = np.array(im)
    im_array = cv2.cvtColor(im_array, cv2.COLOR_BGR2GRAY)
    return app.exec()

if __name__ == "__main__":
    sys.exit(main())
```

**Nuitka call**
```
python3 -m nuitka --onefile --macos-app-icon=icons/icon.icns --macos-app-version=1.6.0 --macos-create-app-bundle --disable-ccache --follow-imports  --enable-plugin=pyside6 --enable-plugin=numpy --assume-yes-for-download --run test.py
```

**Logs**
```
Nuitka-Inclusion:WARNING: Conflicting DLLs for 'libopenjp2.2.5.0.dylib' in your           
Nuitka-Inclusion:WARNING: installation, newest file version used, hoping for the best.
Nuitka-Inclusion:WARNING: Ignoring non-identical DLLs for 'libopenjp2.2.5.0.dylib'.
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/PIL/.dylibs/libopenjp2.2.5.0.dylib
Nuitka-Inclusion:WARNING: used by:
   test.dist/PIL/_imaging.so
different from
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/.dylibs/libopenjp2.2.5.0.dylib
Nuitka-Inclusion:WARNING: used by
   test.dist/cv2/cv2.so
Nuitka-Inclusion:WARNING: Conflicting DLLs for 'liblzma.5.dylib' in your installation,
Nuitka-Inclusion:WARNING: newest file version used, hoping for the best.
Nuitka-Inclusion:WARNING: Ignoring non-identical DLLs for 'liblzma.5.dylib'.
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/PIL/.dylibs/liblzma.5.dylib
Nuitka-Inclusion:WARNING: used by:
   test.dist/PIL/_imaging.so
different from
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/.dylibs/liblzma.5.dylib
Nuitka-Inclusion:WARNING: used by
   test.dist/cv2/cv2.so
Nuitka-Inclusion:WARNING: Conflicting DLLs for 'libwebp.7.dylib' in your installation,
Nuitka-Inclusion:WARNING: newest file version used, hoping for the best.
Nuitka-Inclusion:WARNING: Ignoring non-identical DLLs for 'libwebp.7.dylib'.
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/PIL/.dylibs/libwebp.7.dylib
Nuitka-Inclusion:WARNING: used by:
   test.dist/PIL/_webp.so
different from
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/.dylibs/libwebp.7.dylib
Nuitka-Inclusion:WARNING: used by
   test.dist/cv2/cv2.so
Nuitka-Inclusion:WARNING: Conflicting DLLs for 'libwebpmux.3.dylib' in your installation,
Nuitka-Inclusion:WARNING: newest file version used, hoping for the best.
Nuitka-Inclusion:WARNING: Ignoring non-identical DLLs for 'libwebpmux.3.dylib'.
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/PIL/.dylibs/libwebpmux.3.dylib
Nuitka-Inclusion:WARNING: used by:
   test.dist/PIL/_webp.so
different from
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/.dylibs/libwebpmux.3.dylib
Nuitka-Inclusion:WARNING: used by
   test.dist/cv2/cv2.so
Nuitka-Inclusion:WARNING: Conflicting DLLs for 'libssl.1.1.dylib' in your installation,
Nuitka-Inclusion:WARNING: newest file version used, hoping for the best.
Nuitka-Inclusion:WARNING: Ignoring non-identical DLLs for 'libssl.1.1.dylib'.
Nuitka-Inclusion:WARNING: /Library/Frameworks/Python.framework/Versions/3.8/lib/libssl.1.1.dylib
Nuitka-Inclusion:WARNING: used by:
   test.dist/_hashlib.so
   test.dist/_ssl.so
different
Nuitka-Inclusion:WARNING: from
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/.dylibs/libssl.1.1.dylib
Nuitka-Inclusion:WARNING: used by
   test.dist/cv2/cv2.so
Nuitka-Inclusion:WARNING: Conflicting DLLs for 'libcrypto.1.1.dylib' in your installation,
Nuitka-Inclusion:WARNING: newest file version used, hoping for the best.
Nuitka-Inclusion:WARNING: Ignoring non-identical DLLs for 'libcrypto.1.1.dylib'.
Nuitka-Inclusion:WARNING: /Library/Frameworks/Python.framework/Versions/3.8/lib/libcrypto.1.1.dylib
Nuitka-Inclusion:WARNING: used by:
   test.dist/_hashlib.so
   test.dist/_ssl.so
different
Nuitka-Inclusion:WARNING: from
Nuitka-Inclusion:WARNING: /Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/.dylibs/libcrypto.1.1.dylib
Nuitka-Inclusion:WARNING: used by
   test.dist/cv2/cv2.so
FATAL: Error, problem with dependency scan of 'test.dist/PIL/_imaging.so' with '@loader_path/.dylibs/libtiff.5.dylib' please report the bug.
```

**Note**
I noticed when cv2 is not imported or used at all, the nuitka script works just fine
So if you ran this with the same nuitka options as provided above, it compiles with no issues:

```
"""A simple script to demonstrate the issue"""

import sys
import numpy as np
from PIL import Image
from PySide6.QtCore import Qt
from PySide6.QtWidgets import QApplication, QLabel, QWidget

def main():
    app = QApplication(sys.argv)
    window = QWidget()
    window.setWindowTitle("Simple")
    window.setGeometry(300, 300, 300, 300)
    label = QLabel(window)
    label.setText("Hello World!")
    label.setGeometry(0, 0, 300, 300)
    label.setAlignment(Qt.AlignCenter)
    window.show()

    im = Image.new(mode="RGB", size=(200, 200))
    im_array = np.array(im)
    return app.exec()

if __name__ == "__main__":
    sys.exit(main())
```

**If the `--onefile` option was removed, it returns the following error **
```
File "/Users/User2/Desktop/App-Offline-breakdown3/mainenv/lib/python3.8/site-packages/cv2/__init__.py", line 76, in bootstrap
    raise ImportError('ERROR: recursion is detected during loading of "cv2" binary extensions. Check OpenCV installation.')
ImportError: ERROR: recursion is detected during loading of "cv2" binary extensions. Check OpenCV installation.
```

**Comment by kayhayen** [comment]
I tested OpenC Vsuccessfully the other day, albeit with factory branch, I think the DLL conflict resolution caused this and the merged PR to improve this, had a grave error.

**Comment by statfs** [comment]
cv2 still not work, please help.

**Comment by kayhayen** [comment]
I think it depends on the Python used, I am still working on the DLL resolution not working nicely with Anaconda Python, not sure how many others it affects, but this might take a while still to get right. The original reports and the me toos don't report flavors, so I cannot tell.

**Comment by statfs** [comment]
> I think it depends on the Python used, I am still working on the DLL resolution not working nicely with Anaconda Python, not sure how many others it affects, but this might take a while still to get right. The original reports and the me toos don't report flavors, so I cannot tell.

so which way the the recommended one in MacOS?
1. download python.okg from python.org
2. brew install python( it seems that nuitka rejected the flavor of "Apple Python")
3. local build python from scratch
4. conda install python

**Comment by kayhayen** [comment]
Have you considered checking the User Manual? That's one of the things where it's relatively clear.

**Comment by statfs** [comment]
> Have you considered checking the User Manual? That's one of the things where it's relatively clear.

I am so sorry to trouble you. maybe you are referring to:
 "It has to be CPython, Anaconda Python.
You need the standard Python implementation, called “CPython”, to execute Nuitka, because it is closely tied to implementation details of it."

**Comment by statfs** [comment]
Using the newest version 1.0, the error message is: 
`FATAL: Error, problem with dependency scan of '/opt/miniconda3/envs/py37/lib/python3.7/site-packages/numpy/core/_multiarray_umath.cpython-37m-darwin.so' with '@loader_path/../.dylibs/libopenblas.0.dylib' please report the bug.`

Just FYI.

**Comment by kayhayen** [comment]
That has been reported as a separate issue, strangely I didn't see it myself, you are invited to join #1709 of which your comment is a duplicate I guess, and provide the information I am seeking.

**Comment by kayhayen** [comment]
This has been solved, macOS 1.1.1 hotfix has a severe problem though, but this is unrelated and will be fixed in 1.1.2
